packages/node/src/build.ts

```diff
@@ -143,14 +143,50 @@ async function compile(
   }
 
   let tsCompile: Register;
-  function compileTypeScript(path: string, source: string): string {
+  async function compileTypeScript(
+    path: string,
+    source: string
+  ): Promise<string> {
     const relPath = relative(baseDir, path);
     if (!tsCompile) {
+      // Find package.json to determine module type
+      const pathDir = join(workPath, dirname(path));
+      let pkg: { type?: string } = {};
+
+      // Check if we already have the package.json in cache
+      if (pkgCache.has(pathDir)) {
+        pkg = pkgCache.get(pathDir) || {};
+      } else {
```

Comment:
```suggestion
      let pkg = pkgCache.get(pathDir)

      if (!pkg) {
```

You can simplify this quite a bit